llms/mlx_lm/server.py

```diff
@@ -171,12 +187,18 @@ def validate_model_parameters(self):
         if not isinstance(self.max_tokens, int) or self.max_tokens < 0:
             raise ValueError("max_tokens must be a non-negative integer")
 
+        if isinstance(self.temperature, int):
+            self.temperature = float(self.temperature)
         if not isinstance(self.temperature, float) or self.temperature < 0:
```

Comment:
I think if we just allow integers for these it will be simpler. E.g. 
```suggestion
        if not isinstance(self.temperature, (float, int)) or self.temperature < 0:
```